layers/+lang/python/packages.el

```diff
@@ -226,6 +234,26 @@
         "vps" 'pipenv-shell
         "vpu" 'pipenv-uninstall))))
 
+(defun python/pre-init-pyenv-mode ()
+  (add-to-list 'spacemacs--python-pyenv-modes 'python-mode))
+(defun python/init-pyenv-mode ()
+  (use-package pyenv-mode
+    :if (executable-find "pyenv")
+    :commands (pyenv-mode-versions)
+    :init
+    (pcase python-auto-set-local-pyenv-version
+      ('on-visit
+       (dolist (m spacemacs--python-pyenv-modes)
+         (add-hook (intern (format "%s-hook" m))
+                   'spacemacs//pyenv-mode-set-local-version)))
+      ('on-project-switch
+       (add-hook 'projectile-after-switch-project-hook
+                 'spacemacs//pyenv-mode-set-local-version)))
+    ;; setup shell correctly on environment switch
+    (spacemacs/set-leader-keys-for-major-mode 'python-mode
+      "vu" 'pyenv-mode-unset
+      "vs" 'pyenv-mode-set)))
```

Comment:
Pet is meant to be a drop-in replacement for all the dozens of python virtualenv packages including pyenv. Why would you want to keep pyenv support? Pet already searches for the correct environment for you depending on where you are in your directory.

When pyenv-mode-set is called, pythonic-activate is called and [globally set](https://github.com/pythonic-emacs/pythonic/blob/5daa1b9f589c2faf0ee8165dbb470a4dc752122f/pythonic.el#L293) `python-shell-virtualenv-root`, after this all the [variables](https://github.com/wyuenho/emacs-pet/blob/92da4fa6ffd97fadb24c6fc5d46976606a0264c4/pet.el#L1003-L1005) that depends on it will be out of sync.